index.js

```diff
@@ -20,6 +20,36 @@ StylusFilter.prototype.targetExtension = 'css';
 
 StylusFilter.prototype.processString = function (str) {
 	var opts = this.options;
+
+	var importRegex = new RegExp(/@import ['"].*['"]/);
+	while (importRegex.test(str)) {
+		var imp = importRegex.exec(str);
+		var sourceReg = new RegExp(/['"].*['"]/);
+		var extReg = new RegExp(/.styl/);
+
+		// get import file src and remove quotes
+		var src = sourceReg.exec(imp[0])[0].replace(/"/g, '').replace(/'/g, '');
+
+		// add '.styl' to filename if not exist
+		if (!extReg.test(src)) {
```

Comment:
Why is this done using a regex? Just check the extension using https://github.com/kevva/ext-name?